syntax = "proto3";

package  transport_catalogue_serialize;

import  public  "transport_router.proto";
import  public  "map_renderer.proto";

message Stop {
	string name = 1;
	double lat = 2; // ������
	double lng = 3; // �������
}

message Bus {
	string name = 1;
	repeated string bus_route = 2;
	bool roundtrip  = 3;
}

message Distance {
	string Stop1 = 1;
	string Stop2 = 2;
	uint64 distance = 3;
}

message TransportCatalogue {
	repeated Stop all_stops = 1;
	repeated Bus all_buses = 2;
	repeated  Distance all_distances = 3;
}


message CataloguePack {
	TransportCatalogue catalogue = 1;
	map_renderer_serialize.MapRenderer renderer = 2;
	transport_router_serialize.TransportRouter router = 3;
}